manticore/core/cpu/x86.py

```diff
@@ -3067,6 +3067,78 @@ def PUSHFQ(cpu):
         '''
         cpu.push(cpu.RFLAGS, 64)
 
+    @instruction
+    def PUSHA(cpu):
+        '''
+        Pushes contents of all the registers onto the stack.
+
+        The order in which the register values are pushed into the stack
+        is the same as mentioned in the list below.The value of SP pushed
+        is the value prior to the first push.
+
+        :param cpu: current CPU.
+        '''
+        temp_SP = cpu.read_register('SP')
+        registers = ['AX', 'CX', 'DX', 'BX', temp_SP, 'BP', 'SI', 'DI']
+        for reg in registers:
```

Comment:
This code reads the value of `SP` and uses it as register name, probably not what you intended. Ditto for `PUSHAD`.

Reply:
That was intended behavior. The value of SP pushed into the stack should be the original value of SP before the instruction is executed. That is why I saved the SP value into another variable and pushed it to the stack.

Reply:
The original value of SP isn't what's being pushed however. Try to write a simple test case to confirm these semantics.